Bug 1257818 - [l10n] Build tools: add optional parameter with list of locales to update to update-xliff.py

flodolo opened the pull request:
Replaces #22

st3fan commented on scripts/update-xliff.py:
```diff
@@ -50,15 +50,23 @@ def main():
 
     parser = argparse.ArgumentParser()
     parser.add_argument('base_folder', help='Path to folder including subfolders for all locales')
+    parser.add_argument('locales', default='*', nargs='*', help='Locales to process (default: %(default)s)')
```
This change looks good, I am just a little worried that a argument value of `*` is a bit odd in UNIX terms. It needs to be escaped in the shell otherwise it expands to all files in your current directory.

Or is this not an issue and is `*` just use to recognize the default if the argument is not provided?

flodolo replied:
It's basically used if no argument is provided. Anyway, I've tested this PR on Debian and these commands are equivalent

```
update-xliff.py ~/path/to/l10n-repo
update-xliff.py ~/path/to/l10n-repo *
```

dwaynebailey replied:
@st3fan yeah my fault.  Probably would have been better to check if `locales` is set then use its values, otherwise iterate over all langs.

flodolo replied:
Since this is not merged yet, let's fix it :-)

flodolo commented:
@st3fan @dwaynebailey 
Removed default value, check if `args.locales` is empty instead (it return an empty list if not set). 

Also added a warning if no locales is updated (for example if you provide a non existing path).
